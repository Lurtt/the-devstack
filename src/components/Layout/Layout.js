import React, { Fragment } from 'react'

const Layout = ({ children }) => (
  <Fragment>
    <header>Toolbar</header>
    {children}
    <footer>Footer</footer>
  </Fragment>
)

export default Layout
