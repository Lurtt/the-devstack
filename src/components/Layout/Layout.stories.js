import React from 'react'

import { storiesOf } from '@storybook/react'

import Layout from './'

storiesOf('components', module).add('Layout', () => <Layout />)
