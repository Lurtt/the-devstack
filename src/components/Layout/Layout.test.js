import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Layout from './Layout'

describe('<Layout />', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Layout>Here is the content</Layout>)
  })

  it('should display correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should display header', () => {
    expect(wrapper.find('header').text()).toBe('Toolbar')
  })

  it('should contain two children', () => {
    expect(wrapper.props().children.length).toBe(3)
  })
})
