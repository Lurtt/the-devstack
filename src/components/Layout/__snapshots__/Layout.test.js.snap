// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should display correctly 1`] = `
<React.Fragment>
  <header>
    Toolbar
  </header>
  Here is the content
  <footer>
    Footer
  </footer>
</React.Fragment>
`;
